File: AndroidXPatternsApp/jmvp-app/src/main/java/com/woowrale/jmvp/ui/search/home/SearchActivity.java
package com.woowrale.jmvp.ui.search.home;

import android.content.Intent;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.widget.Toolbar;

import com.woowrale.jmvp.R;
import com.woowrale.jmvp.ui.base.BaseActivity;
import com.woowrale.jmvp.ui.search.local.LocalSearchActivity;
import com.woowrale.jmvp.ui.search.remote.RemoteSearchActivity;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class SearchActivity extends BaseActivity implements SearchView {

    @Inject
    SearchPresenter mPresenter;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initDagger();

        setContentView(R.layout.activity_search);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);
        whiteNotificationBar(toolbar);

        mPresenter.setView(this);

    }

    protected void initDagger() {
        super.getDaggerMainComponent().inject(this);
    }

    @OnClick(R.id.btn_local_search)
    public void openLocalSearch() {
        mPresenter.navigateToSearch("L");
    }

    @OnClick(R.id.btn_remote_search)
    public void openRemoteSearch() {
        mPresenter.navigateToSearch("R");
    }

    private void whiteNotificationBar(View view) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            int flags = view.getSystemUiVisibility();
            flags |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
            view.setSystemUiVisibility(flags);
            getWindow().setStatusBarColor(Color.WHITE);
        }
    }

    @Override
    public void navigateToSearchLocal() {
        startActivity(new Intent(SearchActivity.this, LocalSearchActivity.class));
    }

    @Override
    public void navigateToSearchRemote() {
        startActivity(new Intent(SearchActivity.this, RemoteSearchActivity.class));
    }
}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/mvp/ui/search/home/SearchActivity.kt
package com.woowrale.mvp.ui.search.home

import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.View
import com.woowrale.mvp.R
import com.woowrale.mvp.ui.base.BaseActivity
import com.woowrale.mvp.ui.search.local.LocalSearchActivity
import com.woowrale.mvp.ui.search.remote.RemoteSearchActivity
import kotlinx.android.synthetic.main.activity_search.*
import kotlinx.android.synthetic.main.content_search.*
import javax.inject.Inject

class SearchActivity : BaseActivity(), SearchView {

    @Inject
    lateinit var mPresenter: SearchPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initDagger()

        setContentView(R.layout.activity_search)
        mPresenter.setView(this)

        setSupportActionBar(toolbar)
        whiteNotificationBar(toolbar)

        btnLocalSearch.setOnClickListener({view ->
            mPresenter.navigateToSearch("L")
        })

        btnRemoteSearch.setOnClickListener({view ->
            mPresenter.navigateToSearch("R")
        })
    }

    override fun initDagger() {
        getDaggerMainComponent().inject(this)
    }

    private fun whiteNotificationBar(view: View?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            var flags = view!!.systemUiVisibility
            flags = flags or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
            view.systemUiVisibility = flags
            getWindow().setStatusBarColor(Color.WHITE)
        }
    }

    override fun navigateToSearchLocal() {
        startActivity(Intent(this, LocalSearchActivity::class.java))
    }

    override fun navigateToSearchRemote() {
        startActivity(Intent(this, RemoteSearchActivity::class.java))
    }
}